src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,174 @@
+package duke;
+
+import java.util.ArrayList;
+import java.nio.file.Path;
+import java.io.File;
+import java.io.FileReader;
+import java.io.BufferedReader;
+import java.io.FileWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+
+/**
+ * Represents a Storage that stores tasks.
+ */
+public class Storage {
+    private static int globalIndex = 1;
+
+    /**
+     * Creates an instance of a Storage.
+     * @param filePath Location where the Storage is initialised.
+     */
+    public Storage(Path filePath) {
+        try {
+            if (!java.nio.file.Files.exists(filePath)) { 
+                File newDir = new File("storage" + File.separator + "data.txt");
+                newDir.getParentFile().mkdirs();
+                newDir.createNewFile();
+            }
+        } catch (Exception e) {
+            System.out.println(e);
+        }
+    }
+
+    /**
+     * Copies stored tasks into a list of tasks.
+     * @return ArrayList of stored tasks.
+     */
+    public ArrayList<Task> load() {
+        ArrayList<Task> storedTasks = new ArrayList<>();
+        try {
+            String filePath = "storage" + File.separator + "data.txt";
+            FileReader fr = new FileReader(filePath);
+            BufferedReader br = new BufferedReader(fr);
+            String currentLine;
+            while ((currentLine = br.readLine()) != null) {
```

Comment:
Perhaps it is better to avoid this kind of clever code?
I like it though.

> As the old adage goes, "keep it simple, stupid” (KISS). Do not try to write ‘clever’ code.